src/Microsoft.AspNetCore.Mvc.Razor/TagHelpers/UrlResolutionTagHelper.cs

```diff
@@ -219,23 +219,48 @@ protected void ProcessUrlAttribute(string attributeName, TagHelperOutput output)
         protected bool TryResolveUrl(string url, out string resolvedUrl)
         {
             resolvedUrl = null;
-            if (url == null)
+            if (url == null || url.Length < 2)
             {
                 return false;
             }
 
-            var trimmedUrl = url.Trim(ValidAttributeWhitespaceChars);
+            var maxTestLength = url.Length - 2;
 
-            // Before doing more work, ensure that the URL we're looking at is app-relative.
-            if (trimmedUrl.Length >= 2 && trimmedUrl[0] == '~' && trimmedUrl[1] == '/')
+            var start = 0;
+            for (; start < url.Length; start++)
             {
-                var urlHelper = UrlHelperFactory.GetUrlHelper(ViewContext);
-                resolvedUrl = urlHelper.Content(trimmedUrl);
+                if (start > maxTestLength)
+                {
+                    return false;
+                }
 
-                return true;
+                var i = 0;
+                var ch = url[start];
+                for (; i < ValidAttributeWhitespaceChars.Length; i++)
```

Comment:
Maybe I'm not reading this right - but doesn't this always break (either in the loop or in the `if`?)